### PR TITLE
chore: Document release process

### PR DESCRIPTION
**Pull request by marikaner**
## Context

AI/gen-ai-hub-sdk-js-backlog#133.

Document how to release a new version.

### CODE REVIEW

**Comment by tomfrenken** [comment]
We could also include what to do if any of the expected steps break, what do you think?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -82,6 +82,22 @@ To fix all linting issues, run:
...
+
+To release a new version follow these steps make sure that some prerequisites are met:
+
+- The smoke tests are passing.
+- There are changesets under `.changeset`. Without changesets there should be nothing to release.
+
+If this is the case, follow these steps:
+
+1. **Bump the version**: Execute the `bump` workflow from the `main` branch.
+   If you want to release a new major version, you need to enter the full major version as a precaution.
```
Comment (KavithaSiva) [inline CONTRIBUTING.md:95]:
```suggestion
   If you want to release a new major version, you need to enter the full major version as a precaution.
   Skip providing a version for minor version releases.
```
[question] How does it work for patch? Should we explicitly give the input?

Comment (marikaner) [reply CONTRIBUTING.md:95]: no, the way to you added it is correct for patch versions as well.

**Comment by marikaner** [comment]
> We could also include what to do if any of the expected steps break, what do you think?

@tomfrenken Done, please re-review.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
 $ pnpm lint:fix
 ```
 
+## Releases
+
+To release a new version follow these steps make sure that some prerequisites are met:
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:87]:
```suggestion
To release a new version, ensure that the following prerequisites are met:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+
+- The smoke tests are passing.
+- There are changesets under `.changeset`. Without changesets there should be nothing to release.
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:90]:
```suggestion
- There are changesets under the `.changeset` directory. Without changesets, there should be nothing to release.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+
+If this is the case, follow these steps:
+
+1. **Bump the version**: Execute the `bump` workflow from the `main` branch.
+   If you want to release a new major version, you need to enter the full major version as a precaution.
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:95]:
```suggestion
   If you want to release a new major version, enter the full major version as a precaution.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+   Skip providing a version for minor and patch version releases.
+   This bumps the version on main and creates a tag and draft release with release notes on GitHub.
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:97]:
```suggestion
   This bumps the version on the main branch and creates a tag and draft release with release notes on GitHub.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+2. **Publish to npm**: Find the draft release in the [GitHub releases](https://github.com/SAP/ai-sdk-js/releases), check the release notes and press **Publish release**. This triggers the `publish` workflow that publishes the new version to [`npmjs.com`](https://www.npmjs.com/settings/sap-ai-sdk/packages).
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:98]:
```suggestion
2. **Publish to npm**: Find the draft release in the [GitHub releases](https://github.com/SAP/ai-sdk-js/releases), check the release notes, and press **Publish release**. This triggers the `publish` workflow that publishes the new version to [`npmjs.com`](https://www.npmjs.com/settings/sap-ai-sdk/packages).
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+
+Last, check that everything is published as expected on npm.
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:100]:
```suggestion
Finally, check that everything is published as expected on npm.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+
+### How to Roll Back Releases
+
+Once a release is published on npm you can no longer take it back.
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:104]:
```suggestion
Once a release is published on npm, you can no longer take it back.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+To fix issues in published packages you need to publish a new version.
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:105]:
```suggestion
To fix issues in published packages, you need to publish a new version.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -82,6 +82,37 @@ To fix all linting issues, run:
...
+
+If you find an issue with the release or something fails before publishing, you can revert the release.
+Make sure to:
+
+- Revert the bump commit.
+- Delete the draft release.
+- Delete the version tag.
+
+Then you can fix the error on the state before the release and try again.
```
Comment (ZhongpinWang) [inline CONTRIBUTING.md:114]:
```suggestion
Then you can fix the error and try again from the state before the release.
```

**Review by ZhongpinWang — APPROVED** [review]
Style guide checked with GPT-4o 🤖. Some minor suggestions.

Otherwise, LGTM.

File: CONTRIBUTING.md (hunk 3 above)
Comment (marikaner) [reply CONTRIBUTING.md:90]: I actually noticed that but ignored it anyways 🤫. Thanks for being the style guide police! 🚨

**Pull request merged**